python/cudf/cudf/core/dataframe.py

```diff
@@ -855,10 +860,10 @@ def _from_data(
         data: MutableMapping,
         index: Optional[BaseIndex] = None,
         columns: Any = None,
+        *args,
```

Comment:
Are these `args` & `kwargs` necessary? I see that we are not using it anywhere but is it to compensate for mismatch in method signatures?

Reply:
These are necessary when there is diamond inheritance (which we have for Series from SingleColumnFrame and IndexedFrame) if different method overloads support different subsets of arguments. However, in this case that caused more problems than it solved and I had intended to remove it. I have now removed SingleColumnFrame._from_data, which means that all calls to super()._from_data will use a strict subset of arguments and we can safely remove these from all _from_data implementations.